backend/LexBoxApi/Controllers/LoginController.cs

```diff
@@ -56,6 +58,54 @@ public class LoginController(
         return Redirect(returnTo);
     }
 
+    [HttpPost("signin-google")]
+    [AllowAnonymous]
+    public async Task<ActionResult> GoogleResponse([FromForm] string credential, string? returnTo)
+    {
+        var claimsIdentity = await googleTokenValidator.ValidateGoogleJwt(credential);
+        var email = claimsIdentity?.FindFirst(claim => claim.Type == ClaimTypes.Email);
+        var avatar = claimsIdentity?.FindFirst(claim => claim.Type == "picture");
+        var name = claimsIdentity?.FindFirst(claim => claim.Type == "name");
+        var locale = claimsIdentity?.FindFirst(claim => claim.Type == "locale");
+        ArgumentNullException.ThrowIfNull(email);
+        // ArgumentNullException.ThrowIfNull(avatar);
+        ArgumentNullException.ThrowIfNull(name);
+        // foreach (var claim in claimsIdentity?.Claims ?? [])
+        // {
+        //     Console.WriteLine(claim);
+        // }
+        var userEntity = await userService.GetUserByEmail(email.Value);
+        if (userEntity is null)
+        {
+            var (jwt, _) = lexAuthService.GenerateJwt(new LexAuthUser()
+            {
+                Id = Guid.NewGuid(),
+                Audience = LexboxAudience.RegisterAccount,
+                Name = name.Value,
+                Email = email.Value,
+                EmailVerificationRequired = null,
+                Role = UserRole.user,
+                UpdatedDate = DateTimeOffset.Now.ToUnixTimeSeconds(),
+                CanCreateProjects = null,
+                Locale = locale?.Value ?? LexCore.Entities.User.DefaultLocalizationCode,
+                Locked = null,
+            });
+            var queryParams = new Dictionary<string, string?>() {
+                {"email", email.Value},
+                {"name", name.Value}
+            };
+            var queryString = QueryString.Create(queryParams);
+            Console.WriteLine($"Redirecting to {queryString}...");
+            var redirect = new UriBuilder() { Path = "/register", Query = queryString.ToString() };
+            return Redirect(redirect.ToString());
```

Comment:
this will always redirect to localhost, the UriBuilder seems to default to localhost instead of just making a path. We might be better off not using UriBuilder here.

Reply:
Done in commit b6b4e5a0.